lib/msal-core/src/authority/Authority.ts

```diff
@@ -33,6 +33,25 @@ export class Authority {
         this.tenantDiscoveryResponse = authorityMetadata;
     }
 
+    public static isAdfs(authorityUrl: string): boolean {
+        const components = UrlUtils.GetUrlComponents(authorityUrl);
+        const pathSegments = components.PathSegments;
+
+        if (pathSegments.length && pathSegments[0].toLowerCase() === Constants.ADFS) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public get AuthorityType(): AuthorityType {
+        if (Authority.isAdfs(this.canonicalAuthority)) {
```

Comment:
I assume this is not a ternary because there will be more Authority Types in the future. If that's the case, can we make this a `switch` statement and have AuthorityType.Default be the `default` case?

Reply:
No I actually did make this a ternary but must have lost that change when I was moving stuff around. Thanks for catching this!